core/shared/src/main/scala/zio/ZIOMetric.scala

```diff
@@ -0,0 +1,239 @@
+package zio
+
+import zio.internal.metrics._
```

Comment:
As this, I believe, is a public data type, we should probably move it to `zio.metrics`. Ditto for other public data types associated with metrics.

Reply:
My question here is how many data types we actually need to expose to users. My hypothesis is that the only data type we need to expose to users of metrics is `ZIOMetric` so we could certainly put it in a separate metrics package but if we only have a single data type could also make sense to expose at the top level. Then there could be a couple of other data types necessary for implementers of metrics backend interfaces that could be provided in the `internal` package.